### ISSUE TITLE
Clip Fix (and many other effects) causes all Envelope Points to get shifted/removed - and the envelope is lost with audio reverting to pre-envelope

### ISSUE THREAD

**Issue by bluemoehre** [audacity/audacity#5099]
### Bug description

If having control points on a track and then using Clip Fix will cause all volume points to get moved to the begin of the track. So it is not possible to use this tool at the end of a workflow. 

_**PeterSampson:** many other effects are also afflicted by this issue - see further on in the thread_

Audacity Version 3.3.3

### Steps to reproduce

1. load a heavily compressed clip
2. put some envelope control points (e.g. fade in + fade out)
3. select the clip block
4. choose "Effects" / "Noise Removal & Repair" / "Clip Fix"

### Expected behavior

Clip fix can be applied without touching anything but the audio content of the audio data.

### Actual behavior

All Volume Control Points get shifted to the beginning of the clip

![before](https://github.com/audacity/audacity/assets/1735386/ec5a9d7a-c2da-44e0-8941-eb9b11f51b3c) 
![after](https://github.com/audacity/audacity/assets/1735386/c7461aee-2b4d-46d5-9821-3baff0de3785)

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

### Additional context
See also this bug logged 2 years ago by @crsib :   
**Some effects delete Envelope Control Points, or do not move them when timeline changes #1500**

**Comment by dozzzzer** [comment]
@bluemoehre I cannot reproduce this on either macOS or Windows: with envelop points on the clip and **Clip Fix** applied, I'm seeing those envelope points preserved when applying the effect on a region of the clip (pic 1) or I'm not seeing the envelope points at all when the effect is applied to the whole clip (pic 2):

<img width="550" alt="Screenshot 2023-08-29 at 12 16 05" src="https://github.com/audacity/audacity/assets/70063970/26fec7c4-e656-4887-bf27-7120ef9e8e57">

<img width="550" alt="Screenshot 2023-08-29 at 12 15 36" src="https://github.com/audacity/audacity/assets/70063970/1f00e31c-b82b-49b0-9693-bc2f1a9e71c5">

**Comment by dozzzzer** [comment]
@bluemoehre You might want to provide more details such as the exact settings of the Clip Fix being used, screen recording and the audio file you're working with.

**Comment by petersampsonaudacity** [comment]
@dozzzzer 

I can reproduce this/similar on W10 with not just Clip Fix, but with many other (but not all) effects:
Noise Reduction, Noise Gate, both EQs, all the filters except Bass&Treble, Change Pitch (I have not exhaustively tested)

1. import am audio file
2. crate an envelop point near the beginning at the max level
3. create a second envelope point near the end at a low volume (creates a fade)
4. select all
5. Apply any of the listed effects
6. Observe: corrupted audio

### Before applying effect
![image](https://github.com/audacity/audacity/assets/50205984/00049d2e-bc50-4519-a526-b76c04d62212)

### After applying effect
![image](https://github.com/audacity/audacity/assets/50205984/17924073-1e1c-4bbc-89c4-dd6fa7c1be79)

**Comment by petersampsonaudacity** [comment]
This is a very old regression 2.0.0 has this issue BUT 1.2.6 does not.

So this regression happened sometime in the 1.3.x Beta series (I have yet to analyze this)

**Comment by petersampsonaudacity** [comment]
@dozzzzer 

Since this destroys a users envelope and thus is a data loss surely this should rate **_much_** higher that a lowly P4

**Comment by LWinterberg** [comment]
Fixing envelopes is less of a priority right now because a complete overhaul of envelopes into general automation is one.

**Comment by petersampsonaudacity** [comment]
Full list of affected effects:

![image](https://github.com/audacity/audacity/assets/50205984/258ae7ea-89eb-4c7a-8f92-6bbc73d4406d)

![image](https://github.com/audacity/audacity/assets/50205984/b8052271-9205-4928-a1ef-f83a5d7fbcfe)

**Comment by petersampsonaudacity** [comment]
**Split Stereo** from the TCP also creates an odd result whereby it only retains the envelope fro the track tat was the left channel of the stereo:
![image](https://github.com/audacity/audacity/assets/50205984/ed169256-349c-4e4b-8e33-60f507220916)
But I suspect this is a separate issue

And Swap Stereo Channels from the TCP just remove the envelope entirely.

_I have yet to experiment with making a stereo track from two tracks with mis-matched envelopes_

**Comment by petersampsonaudacity** [comment]
I have added alert divs to the affected effects - here is an example with **Clip Fix**
https://alphamanual.audacityteam.org/man/Clip_Fix

![image](https://github.com/audacity/audacity/assets/50205984/d039eb67-9cd9-4ab5-83d6-add01176819d)

**Comment by bluemoehre** [comment]
@dozzzzer I agree with @petersampsonaudacity - If you don't notice this error in your project right away, it can destroy hours of work. Since we know the problem is caused by 50% of all default filters, this should get a way higher priority.

**Comment by DavidBailes** [comment]
See also: https://github.com/audacity/audacity/issues/1500

**Comment by petersampsonaudacity** [comment]
Today I have added alert and advice divs to the following pages:
1. https://alphamanual.audacityteam.org/man/Envelope_Tool
2. https://alphamanual.audacityteam.org/man/Tools_Toolbar
3. https://alphamanual.audacityteam.org/man/Multi-Tool#Envelope_Tool

In particular this is what is added on the Envelope Tool page;
![image](https://github.com/audacity/audacity/assets/50205984/6eb0d571-0d29-42f2-a39a-c16f35aba689)

and

![image](https://github.com/audacity/audacity/assets/50205984/2ea44897-3ca5-45bb-9dfb-c1596c58a816)

**Comment by literaldehyde** [comment]
Until this is fixed (if it can be fixed), could this be made to trigger a warning popup when users attempt to unwittingly trigger this bug?

**Comment by petersampsonaudacity** [comment]
This is still the case in 3.7.1 and latest 3.7.2 Beta

Any chance this could be fixed for 3.7.2 please as @bluemoehre wrote back in 2023:

>If you don't notice this error in your project right away, it can destroy hours of work. Since we know the problem is caused by 50% of all default filters, this should get a way higher priority.
